torchvision/transforms/_core.py

```diff
@@ -23,17 +24,40 @@ def query_sample(sample: Any, fn: Callable[[Any], Optional[T]]) -> Iterator[T]:
             yield result
 
 
+class TransformDispatch:
+    def __init__(self, transform):
+        self.transform = transform
+        self.transform.set_reset_auto(False)
+
+    def __call__(self, inputs):
+        if isinstance(inputs, collections.abc.Mapping):
+            inputs = {key: self.transform(value) for key, value in inputs.items()}
+        elif isinstance(inputs, collections.abc.Sequence):
+            inputs = tuple(self.transform(input) for input in inputs)
+        self.transform.wipeout_()
+        return inputs
+
+
 class Transform(nn.Module):
     _KNOWN_FEATURE_TYPES = {
         "_".join([part.lower() for part in re.findall("[A-Z][^A-Z]*", feature_type.__name__)]): feature_type
         for feature_type in (Image, BoundingBox)
     }
 
     def __init_subclass__(cls, *, auto_register: bool = True):
+        cls.initialized = False
+        cls.reset_auto = True
         cls._feature_transforms: Dict[Type[Feature], Callable] = {}
         if auto_register:
             cls._auto_register()
 
+    def set_reset_auto(self, mode=True):
+        self.reset_auto = mode
+
+    def wipeout_(self):
```

Comment:
not sure, i can imagine that people will want to play with that in some cases.

Reply:
Can you give examples? Because if there is a need to use these outside of the `JointTransform`, your porposal is more complex / error-prone, because #1 doesn't need these methods and attributes.

Reply:
Frankly it is a rather minor point, we can make them private np